benchmark/storage/parallel_scan_benchmark.cpp

```diff
@@ -0,0 +1,193 @@
+#include "type/transient_value_factory.h"
+#include "type/type_id.h"
+
+namespace terrier {
+class ParalleScanBenchmark : public benchmark::Fixture {};
+
+BENCHMARK_DEFINE_F(ParalleScanBenchmark, TableVectorParallel)(benchmark::State &state) {  // NOLINT
+  // Below is the Working Version
+  uint32_t num_row = 10000000;
+  LoggersUtil::Initialize();
+  auto db_main = terrier::DBMain::Builder().SetUseGC(true).SetUseGCThread(true).SetUseCatalog(true).Build();
+
+  auto block_store = db_main->GetStorageLayer()->GetBlockStore();
+  auto catalog = db_main->GetCatalogLayer()->GetCatalog();
+  auto txn_manager = db_main->GetTransactionLayer()->GetTransactionManager();
+
+  auto test_txn = txn_manager->BeginTransaction();
+
+  // Create catalog and test namespace
+  auto test_db_oid = catalog->CreateDatabase(common::ManagedPointer(test_txn), "test_db", true);
+  auto accessor = catalog->GetAccessor(common::ManagedPointer(test_txn), test_db_oid);
+  auto test_ns_oid = accessor->GetDefaultNamespace();
+  auto exe_ctx = std::make_unique<execution::exec::ExecutionContext>(
+      test_db_oid, common::ManagedPointer(test_txn), nullptr, nullptr, common::ManagedPointer(accessor));
+  execution::sql::TableGenerator table_generator{exe_ctx.get(), block_store, test_ns_oid};
+  table_generator.GenerateBenchmarkTables(false, num_row);
+  auto table_oid = exe_ctx->GetAccessor()->GetTableOid(test_ns_oid, "benchmark_1");
+  struct Counter {
+    uint32_t c_;
+  };
+
+  auto init_count = [](void *ctx, void *tls) { reinterpret_cast<Counter *>(tls)->c_ = 0; };
+
+  // Scan function just counts all tuples it sees
+  auto scanner = [](void *state, void *tls, execution::sql::TableVectorIterator *tvi) {
+    auto *counter = reinterpret_cast<Counter *>(tls);
+    while (tvi->Advance()) {
+      for (auto *pci = tvi->GetProjectedColumnsIterator(); pci->HasNext(); pci->Advance()) {
+        counter->c_++;
+      }
+    }
+  };
+
+  uint32_t col_oids[] = {1, 2};
+
+  // Setup thread states
+  execution::sql::ThreadStateContainer thread_state_container(
+      common::ManagedPointer<execution::sql::MemoryPool>(exe_ctx->GetMemoryPool()));
+  thread_state_container.Reset(sizeof(Counter),  // The type of each thread state structure
+                               init_count,       // The thread state initialization function
+                               nullptr,          // The thread state destruction function
+                               nullptr);         // Context passed to init/destroy functions
+  // NOLINTNEXTLINE
+  for (auto _ : state) {
+    uint64_t elapsed_ms = 0;
+    {
+      common::ScopedTimer<std::chrono::milliseconds> timer(&elapsed_ms);
+      execution::sql::TableVectorIterator::ParallelScan(static_cast<uint32_t>(table_oid),  // ID of table to scan
+                                                        col_oids,  // Query state to pass to scan threads
+                                                        2,         // Container for thread states
```

Comment:
Did the function definition change since this comment? Isn't `2` the `num_oids`?

Reply:
Nice catch! We will fix it